core/src/main/scala/org/allenai/common/FileUtils.scala

```diff
@@ -0,0 +1,73 @@
+package org.allenai.common
+
+import _root_.au.com.bytecode.opencsv.CSVReader
+
+import java.io._
+
+import scala.collection.JavaConverters._
+import scala.io.{ BufferedSource, Codec, Source }
+
+/** Various convenient utilities for reading files and resources. */
+object FileUtils extends Logging {
+
+  /** Get a file as a buffered source. */
+  def getFileAsSource(file: File, codecOpt: Option[Codec] = None): BufferedSource = {
+    codecOpt match {
+      case Some(codec) => Source.fromFile(file)(codec)
+      case None => Source.fromFile(file)
+    }
+  }
+
+  /** Get a file as a sequence of lines. */
+  def getFileAsLines(file: File, codecOpt: Option[Codec] = None): Seq[String] = {
+    logger.debug(s"Loading file ${file.getName}")
+    Resource.using(getFileAsSource(file, codecOpt)) { input =>
+      val lines = input.getLines().toVector // convert to vector to force stream to be processed
+      logger.trace(s"lines:\n\t${lines.mkString("\n\t")}")
+      lines
+    }
+  }
+
+  /** Read a CSV file into a sequence (rows) of sequence (columns) of strings. */
+  def getCSVContentFromFile(file: File): Seq[Seq[String]] = {
+    logger.debug(s"Loading CSV file ${file.getName}")
+    val csvReader = new CSVReader(new FileReader(file))
+    Resource.using(csvReader)(_.readAll.asScala.map(_.toSeq))
+  }
+
+  /** Get a resource file as a Stream. Caller is responsible for closing this stream. */
+  def getResourceAsStream(name: String): BufferedInputStream = {
```

Comment:
I don't think you want these helpers. It obfuscates which type of resource you're loading.

More importantly, you're using the current class to locate the resource, which isn't going to work unless the resource is in the `org.allenai.common` package.

If you want have this take a `Class` instance as a parameter, and clearly document what it's returning, that would be OK.

All in all, I recommend against using resources, because of issues like this. :)